deps/rabbitmq_auth_backend_oauth2/src/oauth_provider.erl

```diff
@@ -0,0 +1,197 @@
+%% This Source Code Form is subject to the terms of the Mozilla Public
+%% License, v. 2.0. If a copy of the MPL was not distributed with this
+%% file, You can obtain one at https://mozilla.org/MPL/2.0/.
+%%
+%% Copyright (c) 2007-2024 Broadcom. All Rights Reserved. The term “Broadcom” refers to Broadcom Inc. and/or its subsidiaries. All rights reserved.
+%%
+
+-module(oauth_provider).
```

Comment:
This is a very generic name, we should continue using `rabbit_oauth2_` for prefix.